File: lab1-circle/java/Circle.java
class Circle {

  public static void main(String[] args) {
    int size = Integer.parseInt(args[0]);
    Circle c = new Circle(size);
    for (int i = 1; i <= size; i++ ) {
      System.out.println(c.circleLine(i));
    }
  }

  public Circle(int size) {
    this.size = size;
  }

  private int size;

  private String circleLine(int i){
    StringBuilder result = new StringBuilder(size);
    for (int jj = 1; jj <= 2*size; jj++) {
      if (shouldPaint(i,jj))
        { result.append('*'); }
      else
        { result.append(' '); }
    }
    return result.toString();
  }

  private Boolean shouldPaint(int i, int jj){
    int j = jj/2;
    return shouldPaintS(size,i,j);
  }

  private Boolean shouldPaintS(int s, int i, int j){
    return Math.abs (Math.pow(i,2) + Math.pow(j,2) - Math.pow(s,2)) <= s+1;
  }
}

File: lab1-circle/python/circle.py
import sys

class Circle:
    def __init__(self, size):
        self.size = size
    def circleLine(self, i):
        result = ""
        for jj in range(1,2*self.size+1):
            if self.shouldPaint(i,jj):
                result += "*"
            else:
                result += " "
        return result
    def shouldPaint(self,i,jj):
        def shouldPaintS(s,i,j): # local function
            return abs (i**2 + j**2 - s**2) <= s+1
        return shouldPaintS(self.size,i,jj/2)

def main():
    size = int(sys.argv[1])
    c = Circle(size)
    for i in range(1,size+1):
        print c.circleLine(i)

main() # interpreter, now that you know what main is, execute it!
